### PR TITLE
DescribeCommand - Add more details on failure

### PR DESCRIPTION
**Pull request by SpacePossum**
I've added the new methods at the end for readability of the PR only.
Besides some clean ups it add suggestion on typos. For example:

```
$ php php-cs-fixer describe @symfony -v

Defined set:
- @PHP56Migration
- @PHP70Migration
- @PHP71Migration
- @PSR1
- @PSR2
- @Symfony
- @Symfony:risky

                                                    
  [InvalidArgumentException]                        
  Set @symfony not found. Did you mean "@Symfony"?  
```

(list of known sets is only shown because I've set the `-v` flag)

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -752,6 +752,7 @@ Exit codes
 Exit code is build using following bit flags:
 
 *  0 OK
+*  1 PHP minimal version to run the Fixer not matched
```
Comment (keradus) [inline README.rst]: nice one. please extract to dedicated PR.

File: src/Console/Command/DescribeCommand.php [hunk 2]
```diff
@@ -204,24 +228,119 @@ private function arrayToText(array $data)
...
+                        $string = preg_replace('/^(\-.*)/', '<fg=red>\1</>', $string);
+                        $string = preg_replace('/^(@.*)/', '<fg=cyan>\1</>', $string);
+
+                        return '   '.$string;
+                    },
+                    preg_split("#\n\r|\n#", OutputFormatter::escape(rtrim($diff)))
+                )
+            );
+        } else {
+            $template = '      ---------- begin diff ----------%s      ----------- end diff -----------';
```
Comment (keradus) [inline src/Console/Command/DescribeCommand.php]: no no no no no, I initially remove that checking `isDecoratedOutput`. it's not needed and it looks worse when you change the template.

File: src/Console/Command/DescribeCommand.php [hunk 3]
```diff
@@ -204,24 +228,119 @@ private function arrayToText(array $data)
...
+     * @return string|null
+     */
+    private function getAlternative($type, $name)
+    {
+        $other = null;
+        $alternatives = 'set' === $type ? $this->getSetNames() : $this->getFixers();
+
+        foreach ($alternatives as $alternative) {
+            $distance = levenshtein($name, $alternative);
+            if ($distance < 2) {
```
Comment (keradus) [inline src/Console/Command/DescribeCommand.php]: btw, why 2 ?

Comment (SpacePossum) [inline src/Console/Command/DescribeCommand.php]: I think SF default is 3 for commands, its all about feeling I guess between false positive and missed opportunity.

Comment (keradus) [inline src/Console/Command/DescribeCommand.php]: oh, yoda ;P

File: src/Console/Command/DescribeCommand.php (hunk 2 above)
Comment (SpacePossum) [inline src/Console/Command/DescribeCommand.php]: alright

File: README.rst (hunk 1 above)
Comment (SpacePossum) [inline README.rst]: I'll put it on my list, remind me before 2.x gets shipped if I didn't do this

**Comment by keradus** [comment]
Thank you @SpacePossum
